### PR TITLE
infrastructure-agent: init at 1.57.2

### PR DESCRIPTION
**Pull request by DavSanchez**
Add the New Relic Infrastructure Agent derivation.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/in/infrastructure-agent/package.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  CGO_ENABLED = if stdenv.isDarwin then "1" else "0";
+
+  subPackages = [
+    "cmd/newrelic-infra"
+    "cmd/newrelic-infra-ctl"
+    "cmd/newrelic-infra-service"
+  ];
+
+  meta = with lib; {
```
Comment (stv0g) [inline pkgs/by-name/in/infrastructure-agent/package.nix:35]: Please avoid `with lib`.

See: https://nix.dev/guides/best-practices.html#with-scopes

Comment (DavSanchez) [reply pkgs/by-name/in/infrastructure-agent/package.nix]: Thanks! Done for the `meta` attrset, but still using it for `maintainers` as more could be added to the list. Let me know if it's good practice to still not use it e.g. when the lists have few items.

File: pkgs/by-name/in/infrastructure-agent/package.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    "cmd/newrelic-infra"
+    "cmd/newrelic-infra-ctl"
+    "cmd/newrelic-infra-service"
+  ];
+
+  meta = {
+    description = "New Relic Infrastructure Agent";
+    homepage = "https://github.com/newrelic/infrastructure-agent.git";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ DavSanchez ];
```
Comment (FliegendeWurst) [inline pkgs/by-name/in/infrastructure-agent/package.nix:39]:
```suggestion
    maintainers = with lib.maintainers; [ davsanchez ];
```

File: pkgs/by-name/in/infrastructure-agent/package.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  vendorHash = "sha256-0WLL15CXRi/flp4EV3Qt0wO1VaUmAokzsChpiqjs+YQ=";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.buildVersion=${version}"
+    "-X main.gitCommit=${src.rev}"
+  ];
+
+  CGO_ENABLED = if stdenv.isDarwin then "1" else "0";
```
Comment (FliegendeWurst) [inline pkgs/by-name/in/infrastructure-agent/package.nix:27]:
```suggestion
  CGO_ENABLED = if stdenv.hostPlatform.isDarwin then "1" else "0";
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/in/infrastructure-agent/package.nix (hunk 3 above)
Comment (DavSanchez) [reply pkgs/by-name/in/infrastructure-agent/package.nix]: Done!

**Review by Aleksanaa — APPROVED** [review]
Tested on aarch64-darwin

**Pull request merged**